### ISSUE TITLE
Customized YOLOv8-OBB model performed well in validation set, but detect zero object in testing set.

### ISSUE THREAD

**Issue by lixiaoqingnnz** [ultralytics/ultralytics#14591]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have trained a YOLOv8-OBB model from scratch to detect the digitals in the blood pressure machines. 

**Preprocessed:**
- Annotated and exported the dataset using Roboflow, which should not have problem of overlapping the dataset.

**Code**
```
from ultralytics import YOLO

# train 
model = YOLO("yolov8-obb.pt")
model = YOLO("data.yaml")
results = model.train(data="coco8.yaml", epochs=100, imgsz=640, device=[0, 1])

# val
trained_model = YOLO("path/to/last.pt")  # load a partially trained model
metrics = trained_model.val()

# test
result = trained_model("5.jpg")
```

**Results:**
<img width="683" alt="image" src="https://github.com/user-attachments/assets/8a02ebce-7875-4875-ab42-fd5863a91ddc">
Results on validation set looks fine. The left picture shows the labels and the right picture shows the prediction. 

**Problem:**
However, when I used the image in test set to test the model, the model can not detect anything. As shown in the below picture, it detected 0 bounding box. I have tried different image, but the model just can not detect anything. 
![image](https://github.com/user-attachments/assets/de64d145-c331-4831-9172-507a8d5a8835)

Has anyone encountered similar situation before? Appreciate for any help!

**Comment by Y-T-G** [comment]
What's the actual code for prediction? The one you posted doesn't make sense.

**Comment by lixiaoqingnnz** [comment]
> What's the actual code for prediction? The one you posted doesn't make sense.

Sorry the code was not completed. I just edited, pls check.

**Comment by Y-T-G** [comment]
Did you use `results.plot()` to check?

**Comment by Y-T-G** [comment]
or add `show=True`

**Comment by lixiaoqingnnz** [comment]
> or add `show=True`

Hi I use `result.show()` to check and find out that the model works well. Why I didn't get the result at the beginning is because:

1. Though I used one image to test, `result = trained_model("5.jpg")` returns a list of the results, so `result = result[0]` is required to check the result I want.
2. Because yolov8-obb returns ultralytics.engine.results.OBB object, so there are no boxes shown in the result. 

Thank you so much for your help^^

**Comment by Y-T-G** [comment]
For obb, you access results with `result.obb`

**Comment by Laughing-q** [comment]
@lixiaoqingnnz Closing the issue since it seems to be resolved. :) Please feel feel to reopen it if you encounter any other related issues.
